### PR TITLE
Ensure CSINode allocatable count is set on node before removing startup taint

### PR DESCRIPTION
**Pull request by torredil**
**Is this a bug fix or adding new feature?**

Bug Fix

**What is this PR about? / Why do we need it?**

Fixes https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1945

**What testing is done?** 

manual testing && `make test` && e2e

### CODE REVIEW

File: pkg/driver/node.go [hunk 1]
```diff
@@ -103,7 +106,9 @@ func newNodeService(driverOptions *DriverOptions) nodeService {
 
 	// Remove taint from node to indicate driver startup success
 	// This is done at the last possible moment to prevent race conditions or false positive removals
-	go removeTaintInBackground(cloud.DefaultKubernetesAPIClient, removeNotReadyTaint)
+	time.AfterFunc(taintRemovalInitialDelay, func() {
```
Comment (ConnorJC3) [inline pkg/driver/node.go:109]: Any particular reason to add a 1 second delay? Wasn't mentioned in PR description or original bug

File: pkg/driver/node_test.go [hunk 2]
```diff
@@ -2414,16 +2445,52 @@ func TestRemoveNotReadyTaint(t *testing.T) {
...
+								NodeID: nodeName,
+								Allocatable: &v1.VolumeNodeResources{
+									Count: &count,
+								},
+							},
+						},
+					},
+				}
+
+				csiNodesMock.EXPECT().
```
Comment (ConnorJC3) [inline pkg/driver/node_test.go:2485]: I know that I am the one responsible for it, and I'm not asking you to fix what's already broken in this PR.

But in the future, we really need to move from this boilerplate disaster to using the fake client implementation.

File: pkg/driver/node.go (hunk 1 above)
Comment (torredil) [reply pkg/driver/node.go:109]: Tested various intervals, first call expected failures observed at 900ms, none at 1 second.

**Review by AndrewSirenko — APPROVED** [review]

**Pull request merged**